src/Tokenizer.ts

```diff
@@ -586,10 +600,10 @@ export class Tokenizer {
     }
   }
 
-  reflink(src, links) {
+  reflink(src: string, links: Record<string, Pick<Tokens.Link | Tokens.Image, 'href' | 'title'>>): Tokens.Link | Tokens.Image | Tokens.Text | undefined {
```

Comment:
I extracted the type into a separate type (still using `Pick`). I thought it would be better because it ensures the types stay in sync and will yell at you when you make changes to `Tokens.Image` that are incompatible with what reflink expects. But if you'd prefer I can also make it its own type.

Reply:
I would personally prefer typing all the different Tokens in the same way, in the same place, for the sake of future maintainability.

Yes, the two types wouldn't "sync", but at least we don't have this one type definition hiding in here where all the others are together in a different file.

Reply:
I think the token types should stay consistent across different tokenizers so when someone uses walktokens and sees `token.type === "image"` they don't have to care if that image token was created by a reflink or image tokenizer.

If we want the reflink token to be different from image we should have a "reflink" type.

Reply:
@Uzitech Here I'm talking about the typescript type definition, not the Token.type object property.

Reply:
Are you talking about the return type of reflink (`Tokens.Link | Tokens.Image | Tokens.Text | undefined`) or the type of the links param (`Record<string, Pick<Tokens.Link | Tokens.Image, 'href' | 'title'>>`)?

Reply:
I'm talking about `Record<string, Pick<Tokens.Link | Tokens.Image, 'href' | 'title'>>` on the links param. Perhaps I misunderstood what was going on. But it just seemed weird to create a type on the fly here where all our other Token-related types are in `src/Tokens.ts`.

Reply:
Like so? https://github.com/markedjs/marked/pull/2805/commits/48db511a35da04ea5d593d5767b385df7d2b91c7